### PR TITLE
Chylli/replace datetime

### PR DESCRIPTION
**Pull request by chylli**

### CODE REVIEW

File: lib/Date/Utility.pm [hunk 1]
```diff
@@ -1015,6 +1041,115 @@ sub today {
...
+
+    print Date::Utility->new('2000-02-29')->_plus_years(1)->date_yyyymmdd;
+    # will got 2001-02-28
+
+=cut
+
+sub _plus_years {
+    my ($self, $years) = @_;
+    (looks_like_number($years) && $years == int($years)) || die "Need a integer years number";
+    return _create_trimmed_date($self, $self->year + $years, $self->month, $self->day_of_month);
```
Comment (basiliscos) [inline lib/Date/Utility.pm]: it looks you are loosing hour, minutes, seconds in that method. Are you sure, that this is required? 

I.e. I 'd expect that `->new("2018-02-04 10:45:22")->plus("1y")` will be `2019-02-04 10:45:22` and not `2019-02-04 00:00:00`.

File: lib/Date/Utility.pm [hunk 2]
```diff
@@ -1015,6 +1041,115 @@ sub today {
...
+    if ($new_month < 1 || $new_month > 12) {
+        $new_year += floor($new_month / 12);
+        $new_month = $new_month % 12;
+        while ($new_month < 1) {
+            $new_year--;
+            $new_month += 12;
+        }
+    }
+    my $new_day = $self->day_of_month;
+    return _create_trimmed_date($self, $new_year, $new_month, $new_day);
```
Comment (basiliscos) [inline lib/Date/Utility.pm]: the same comment as for years above.

File: lib/Date/Utility.pm (hunk 1 above)
Comment (chylli) [reply lib/Date/Utility.pm]: Please see the line https://github.com/binary-com/perl-Date-Utility/pull/23/files#diff-9982e44798acaa1521793b3f10130e02R1127 , the time part is in `$self` object.

Comment (chylli) [reply lib/Date/Utility.pm]: @basiliscos Maybe can you help me to give the function `_create_trimmed_date` a more meaningful name ?

Comment (basiliscos) [reply lib/Date/Utility.pm]: @chylli `_override($y, $m, $d, $h, $m, $s)` ?

Comment (chylli) [reply lib/Date/Utility.pm]: No, what I want to express is : `If the day is greater than the max day in that month , then use that max day as the day in the new object.`

**Pull request merged**